gtars/src/uniwig/mod.rs

```diff
@@ -1,16 +1,722 @@
+                }
+
+            }
+
+            if coordinate_position%stepsize == 0{
+                // Step size defaults to 1, so report every value
+                v_coord_counts.push(count);
+                v_coordinate_positions.push(coordinate_position);
+                //println!("DEBUG: Reporting count: {} at position: {} for adjusted start site: {}",count, coordinate_position, adjusted_start_site);
+
+            }
+
+            //println!("DEBUG: Incrementing coordinate_position: {}  -> {}", coordinate_position,  coordinate_position +1);
+            coordinate_position = coordinate_position + 1;
+
+
+        }
+
+        prev_coordinate_value = adjusted_start_site;
+
+    }
+
+    count = count + 1; // We must add 1 extra value here so that our calculation during the tail as we close out the end sites does not go negative.
+    // this is because the code above subtracts twice during the INITIAL end site closure. So we are missing one count and need to make it up else we go negative.
+    //
+
+    while coordinate_position <= chrom_size + 1 + smoothsize*2{ // Apply an bound to push the final coordinates otherwise it will become truncated.
+
+        while current_end_site==coordinate_position{
+
+            count = count - 1;
+
+            if collected_end_sites.last() == None {
+                current_end_site = 0; // From original code. Double check this is the proper way.
+            } else {
+                current_end_site = collected_end_sites.remove(0)
+            }
+
+        }
+
+        if coordinate_position % stepsize == 0{
+            // Step size defaults to 1, so report every value
+            v_coord_counts.push(count);
+            v_coordinate_positions.push(coordinate_position); // This is ONLY the starts
+            //println!("DEBUG: Reporting count: {} at start position: {} and end position: ",count, coordinate_position);
+
+        }
+
+        //println!("DEBUG: Incrementing coordinate_position: {}  -> {}", coordinate_position,  coordinate_position +1);
+        coordinate_position = coordinate_position + 1;
+
+
+    }
+
+
+    //println!("DEBUG: FINAL LENGTHS... Counts: {:?}  Positions: {:?}", v_coord_counts, v_coordinate_positions);
+    return (v_coord_counts, v_coordinate_positions)
+}
+#[allow(unused_variables)]
+pub fn fixed_core_wiggle(starts_vector: &Vec<i32>, ends_vector: &Vec<i32>, chrom_size: i32, stepsize:i32) -> (Vec<u32>, Vec<i32>) {
+    // This function is a more direct port of fixedCoreBW from uniwig written in CPP
+    // It allows the user to accumulate reads of across paired starts and ends.
+    // Counts occur between a start coordinate (cutSite) and an end site (endSite) where the endsite is determined based on
+    // the paired ends.
+    // Counts are reported over a stepsize (with a default of stepsize = 1)
+    // Unlike the original function, it does not write to disk in chunks. it simply returns a vector of accumulated reads.
```

Comment:
If you move these above the function header... I believe they get pulled into `docs.rs` and will appear in type hints in peoples IDEs

Reply:
great idea. Done!